ansible/roles/invoker/tasks/deploy.yml

```diff
@@ -3,16 +3,40 @@
 
 - include: docker_login.yml
 
-- name: "pull action images with tag {{docker_image_tag}}"
+- name: "pull invoker images with tag {{docker_image_tag}}"
   shell: "docker pull {{docker_registry}}{{item}}:{{docker_image_tag}}"
   with_items:
     - '{{ docker_image_prefix }}/invoker'
-    - '{{ docker_image_prefix }}/dockerskeleton'
-    - '{{ docker_image_prefix }}/nodejs6action'
-    - '{{ docker_image_prefix }}/python2action'
-    - '{{ docker_image_prefix }}/python3action'
-    - '{{ docker_image_prefix }}/swift3action'
-    - '{{ docker_image_prefix }}/java8action'
+  when: docker_registry != ""
+
+- name: "pull nodejs action images with tag {{docker_image_tag}}"
+  shell: "docker pull {{docker_registry}}{{item.image.name}}:{{docker_image_tag}}"
+  with_items:
+    - "{{ runtimesManifest.runtimes.nodejs }}"
```

Comment:
I worked with @dubeejw to see if there was a way to not duplicate here and be able to itirate over the dict, and then iterate over the array. we tried different combinations and were not successful.

If there is a ansible ninja that would provide how to achieve this, from the docs it gives the impression is doable but we didn't found a good combination.
http://docs.ansible.com/ansible/playbooks_loops.html